extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -93,6 +93,21 @@ public CompletionStage<V> get() {
                 });
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K, V> Uni<V> getAsync(K key, Function<K, Uni<V>> valueLoader) {
+        Objects.requireNonNull(key, NULL_KEYS_NOT_SUPPORTED_MSG);
+        return (Uni<V>) Uni.createFrom()
+                .completionStage(cache.asMap().computeIfAbsent(key, new Function<Object, CompletableFuture<Object>>() {
+                    @Override
+                    public CompletableFuture<Object> apply(Object key) {
+                        return (CompletableFuture<Object>) valueLoader.apply((K) key)
```

Comment:
I'm not so sure about exception handling, e.g. when the `loader.apply()` throws an unchecked exception should we wrap it in a `CacheException`?

Reply:
Probably not because for `Cache#get()` we only wrap the checked exceptions.

Reply:
Glad your alter ego got his question answered 😆